benchmark/testsystems/bookkeepers.py

```diff
@@ -45,138 +45,114 @@ def accumulate_shadow_work(self, x_0, v_0, n_steps):
-        for _ in tqdm(range(100)):
-            min_sim.step(1)
-        pos = get_positions(min_sim)
-        del (min_sim)
-        gc.collect()
-
-        # "Equilibrate" / "burn-in"
-        # Running a bit of Langevin first improves GHMC acceptance rates?
-        print("Intializing with Langevin dynamics...")
-        langevin_sim = self.construct_simulation(
-            LangevinSplittingIntegrator("V R O R V", temperature=self.temperature,
-                                        collision_rate=1.0 / unit.picoseconds,
-                                        timestep=2 * self.ghmc_timestep, measure_heat=False)
-        )
-        set_positions(langevin_sim, pos)
-        for _ in tqdm(range(self.burn_in_length)):
-            langevin_sim.step(1)
-        pos = get_positions(langevin_sim)
-        del (langevin_sim)
-        gc.collect()
-
-        print('"Burning in" unbiased GHMC sampler for {:.3}ps...'.format(
-            (self.burn_in_length * self.ghmc_timestep).value_in_unit(unit.picoseconds)))
-        self.unbiased_simulation = self.construct_simulation(
-            GHMCIntegrator(temperature=self.temperature, timestep=self.ghmc_timestep))
-        set_positions(self.unbiased_simulation, pos)
+        print('"Burning in" unbiased sampler for {:.3}ps...'.format(
+            (self.burn_in_length * self.xcghmc_timestep * 10).value_in_unit(unit.picoseconds)))
         for _ in tqdm(range(self.burn_in_length)):
             self.unbiased_simulation.step(1)
-        print("Burn-in GHMC acceptance rate: {:.3f}%".format(100 * self.get_ghmc_acceptance_rate()))
-        self.reset_ghmc_statistics()
+        print("Burn-in XC-GHMC acceptance rate: {:.3f}%".format(100 * self.get_acceptance_rate()))
 
         # Collect equilibrium samples
         print("Collecting equilibrium samples...")
         equilibrium_samples = []
+        box_vectors = []
         for _ in tqdm(range(self.n_samples)):
             self.unbiased_simulation.step(self.thinning_interval)
-            x = self.unbiased_simulation.context.getState(getPositions=True).getPositions(asNumpy=True)
+            state = self.unbiased_simulation.context.getState(getPositions=True)
+            x = state.getPositions(asNumpy=True)
             equilibrium_samples.append(strip_unit(x))
-        print("Equilibrated GHMC acceptance rate: {:.3f}%".format(100 * self.get_ghmc_acceptance_rate()))
+            box_vectors.append(state.getPeriodicBoxVectors())
+        print("Equilibrated XC-GHMC acceptance rate: {:.3f}%".format(100 * self.get_acceptance_rate()))
 
-        return np.array(equilibrium_samples)
+        return np.array(equilibrium_samples), np.array(box_vectors)
 
-    def get_path_to_samples(self):
+    def get_path_to_samples_and_box_vectors(self):
         """Samples are {name}_samples.npy in DATA_PATH"""
-        return os.path.join(DATA_PATH, '{}_samples.npy'.format(self.name))
+        samples = os.path.join(DATA_PATH, '{}_samples.npy'.format(self.name))
```

Comment:
It would seem much safer/easier to just store both `samples` and `box_vectors` in one `.npy` file. I believe `.npy` files support storage of multiple numpy objects.